cloudbuild/python-scripts/parse_logs.py

```diff
@@ -0,0 +1,281 @@
+"""Python Script to parse yarn logs to check of job success.
+
+This python file extracts the status of a dataproc job from its job-id and then
+uses this job-id to get the yarn application number.
+The yarn application number enables it to read through the yarn logs to check
+for the number iof records.
+In case the number of records match that of BQ table it returns, in case of
+mismatch, it throws an error.
+"""
+
+from collections.abc import Sequence
+
+from absl import app
+from google.cloud import bigquery
+from google.cloud import dataproc_v1
+from google.cloud import storage
+import re
+
+
+def get_bq_query_rows(client_project_name):
+    client = bigquery.Client(project=client_project_name)
+    query = 'SELECT count(*) as count FROM `testproject-398714.testing_dataset.largeTable` where EXTRACT(HOUR from ts) = 17 and EXTRACT(DAY from ts) = 17;'
+    query_job = client.query(query)
+    query_result = query_job.result()
+    records_read = 0
+    for result in query_result:
+        records_read = result[0]
+    return records_read
+
+# Remember these are the ones from args.
+def get_bq_table_rows(client_project_name, project_name, dataset_name, table_name, query):
+    if query:
+        return get_bq_query_rows(client_project_name)
+    dataset_ref = bigquery.DatasetReference(project=project_name, dataset_id=dataset_name)
+    table_ref = bigquery.TableReference(dataset_ref=dataset_ref, table_id=table_name)
+    client = bigquery.Client(project=client_project_name)
+    table = client.get_table(table=table_ref)
+    row_count = table.num_rows
+    return row_count
+
+def extract_metric(logs_as_string, metric_string, delimiter):
+    total_metric_sum_in_blob = 0
+    # Keep on finding the metric value as there can be
+    # 1 or more outputs in a log file.
+    metric_pattern = r'{}\s*(.*?)\s*{}'.format(re.escape(metric_string), re.escape(delimiter))
+    metric_pattern = re.compile(metric_pattern)
+    matches = metric_pattern.finditer(logs_as_string)
+    records_read = [int(record_read.group(1).strip()) for record_read in matches]
+    total_metric_sum_in_blob += sum(records_read)
+    return total_metric_sum_in_blob
+
+def check_query_correctness(logs_as_string):
+    query_records_pattern = r'\[\s(.*),\s(.*)\s\]'
+    # Find all matches of the pattern in the string
+    matches = re.findall(query_records_pattern, logs_as_string)
+    # Extract and print all pairs of HOUR and DAY
+    if matches:
+        for match in matches:
+            hour = match[0].strip()
+            day = match[1].strip()
+            if hour !='17' or day !='17':
```

Comment:
Please mention that these checks are hardcoded to check the filter query from above